setup.py

```diff
@@ -118,11 +118,8 @@ def run(self):
         # Construct library name based on ext suffix (contains python version, architecture and .so)
         library_name = "libcustom_calls.so"
 
-        package_root = path.dirname(__file__)
-        frontend_path = glob.glob(
-            path.join(package_root, "frontend", "**", library_name), recursive=True
-        )
-        build_path = glob.glob(path.join("build", "**", library_name), recursive=True)
+        frontend_path = Path("frontend").absolute().glob(f"**/{library_name}")
+        build_path = Path("build").absolute().glob(f"**/{library_name}")
         lib_with_r_path = "@rpath/libcustom_calls.so"
 
         original_path = frontend_path[0] if frontend_path else build_path[0]
```

Comment:
not working on macOS :(

Check the issues here,
- https://github.com/PennyLaneAI/catalyst/actions/runs/8070726162/job/22049054457?pr=551
- https://github.com/PennyLaneAI/catalyst/actions/runs/8070726160/job/22049055423?pr=551